mesonbuild/coredata.py

```diff
@@ -595,6 +598,24 @@ def __init__(self, options: argparse.Namespace, scratch_dir: str, meson_command:
         self.builtin_options_libdir_cross_fixup()
         self.init_builtins('')
 
+    @staticmethod
+    def __parse_program_aliases(options: argparse.Namespace, for_machine: MachineChoice) -> OrderedDict[str, str]:
+        alias_map: OrderedDict[str, str] = OrderedDict()
+        aliases: T.List[str]
+        if for_machine == MachineChoice.HOST:
+            aliases = options.native_program or []
+        elif for_machine == MachineChoice.BUILD:
+            aliases = options.cross_program or []
+
+        for a in aliases:
+            parts = a.split('=', 1)
+            if len(parts) == 1:
+                pass
+
+            alias_map[parts[0]] = parts[1]
```

Comment:
What about
```suggestion
        alias_map.update(p.split('=', maxsplit=1) for p in parts if '=' in p)
```

Reply:
Wouldn't this call update needlessly?

Reply:
Huh? What do you mean? It iterates through aliases, test for the presence of `=`, split it, and add it to the dict, just like your for loop...

Reply:
The difference is that when aliases is empty, a for loop doesn't enter the block body, while `alias_map.update(generator_expression)` calls the `.update()` method with no values.

That being said, it's a good point to do:
```python
for a in aliases:
    if '=' in a:
        parts = a.split('=', 1)
        alias_map[parts[0]] = parts[1]
```